code/iaas/ha/src/main/java/io/cattle/platform/ha/monitor/impl/PingInstancesMonitorImpl.java

```diff
@@ -36,13 +52,30 @@
 
     private static final DynamicLongProperty CACHE_TIME = ArchaiusUtil.getLong("ha.instance.state.cache.millis");
 
+    private static final Logger log = LoggerFactory.getLogger(PingInstancesMonitorImpl.class);
+
+    private static final String UKNOWN_OUT_OF_SYNC_WARNING = "Instance out of sync and can't determine action to take. Uuid [{}]. Docker id [{}]. "
+            + "State in rancher [{}]. State on host [{}]";
+
+    @Inject
+    AgentDao agentDao;
+    @Inject
+    ObjectMetaDataManager objectMetaDataManager;
+    @Inject
+    AgentLocator agentLocator;
+    @Inject
+    LockDelegator lockDelegator;
+    @Inject
     PingInstancesMonitorDao monitorDao;
+    @Inject
     ObjectManager objectManager;
+    @Inject
     ObjectProcessManager processManager;
-    LoadingCache<Long, Map<String, String>> instanceCache = CacheBuilder.newBuilder().expireAfterWrite(CACHE_TIME.get(), TimeUnit.MILLISECONDS).build(
-            new CacheLoader<Long, Map<String, String>>() {
+
+    LoadingCache<Long, Map<String, Instance>> instanceCache = CacheBuilder.newBuilder().expireAfterWrite(CACHE_TIME.get(), TimeUnit.MILLISECONDS)
```

Comment:
@ibuildthecloud Note that I'm now putting instances in this cache. Think they are too big memory-wise? If so, I could change to a custom pojo that has just the fields I need (would be four or five fields).